File: Awesome.AI.Source/Awesome.AI/Interfaces/IMechanics.cs
using Awesome.AI.Awesome.AI.Core;
using Awesome.AI.Core;
using Awesome.AI.Core.Mechanics;
using static Awesome.AI.Variables.Enums;

namespace Awesome.AI.Interfaces
{
    public interface IMechanics
    {
        MECHANICS type { get; set; }
        MechSymbolicOut ms { get; set; }
        MechParams mp { get; set; }

        double POS_XY { get; }

        void Peek(UNIT c);

        //these are thought patterns: general, good, bad
        void Calculate(PATTERN match, int cycles);
    }
}

File: Awesome.AI.Python/Interfaces/IMechanics.py
from abc import ABC, abstractmethod


class IMechanics(ABC):
    @abstractmethod
    def PosXY(self) -> float: ...

    @abstractmethod
    def Calc(self, curr, cycles: int) -> None: ...

    @abstractmethod
    def Calculate(self, match, cycles: int) -> None: ...

